ddtrace/tracer/sqlcomment.go

```diff
@@ -0,0 +1,135 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package tracer
+
+import (
+	"fmt"
+	"net/url"
+	"sort"
+	"strconv"
+	"strings"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/globalconfig"
+)
+
+// SQLCommentInjectionMode represents the mode of sql comment injection.
+type SQLCommentInjectionMode int
+
+const (
+	// SQLInjectionDisabled represents the comment injection mode where all injection is disabled.
+	SQLInjectionDisabled SQLCommentInjectionMode = 0
+	// SQLInjectionModeService represents the comment injection mode where only service tags (name, env, version) are injected.
+	SQLInjectionModeService SQLCommentInjectionMode = 1
+	// SQLInjectionModeFull represents the comment injection mode where both service tags and tracing tags. Tracing tags include span id, trace id and sampling priority.
+	SQLInjectionModeFull SQLCommentInjectionMode = 2
+)
+
+// Key names for SQL comment tags.
+const (
+	sqlCommentKeySamplingPriority = "ddsp"
+	sqlCommentTraceID             = "ddtid"
+	sqlCommentSpanID              = "ddsid"
+	sqlCommentService             = "ddsn"
+	sqlCommentVersion             = "ddsv"
+	sqlCommentEnv                 = "dde"
+)
+
+// SQLCommentCarrier is a carrier implementation that injects a span context in a SQL query in the form
+// of a sqlcommenter formatted comment prepended to the original query text.
+// See https://google.github.io/sqlcommenter/spec/ for more details.
+type SQLCommentCarrier struct {
+	Query  string
+	Mode   SQLCommentInjectionMode
+	SpanID uint64
+}
+
+// Inject injects a span context in the carrier's query.
+func (c *SQLCommentCarrier) Inject(spanCtx ddtrace.SpanContext) error {
+	c.SpanID = random.Uint64()
+	tags := make(map[string]string)
+	switch c.Mode {
+	case SQLInjectionDisabled:
+		return nil
+	case SQLInjectionModeFull:
+		samplingPriority := 0
+		var traceID uint64
+		ctx, ok := spanCtx.(*spanContext)
+		if ok {
+			if sp, ok := ctx.samplingPriority(); ok {
+				samplingPriority = sp
+			}
+			if ctx.TraceID() > 0 {
+				traceID = ctx.TraceID()
+			}
```

Comment:
```suggestion
			traceID = ctx.TraceID()
```
It shouldn't matter, right? `traceID` is `0` here anyway. Or do you think we'd be opening up to a vulnerability from future changes if we do that?